code/game/gamemodes/vampire/traitor_vamp.dm

```diff
@@ -19,20 +21,26 @@
 		restricted_jobs += protected_jobs
 
 	var/list/datum/mind/possible_vampires = get_players_for_role(ROLE_VAMPIRE)
+	secondary_enemies = CEILING((secondary_enemies_scaling * num_players()), 1)
 
 	for(var/mob/new_player/player in GLOB.player_list)
-		if((player.mind in possible_vampires) && (player.client.prefs.species in protected_species_vampire))
+		if((player.mind in possible_vampires) && (player.client.prefs.species in secondary_protected_species))
 			possible_vampires -= player.mind
 
 	if(possible_vampires.len > 0)
-		var/datum/mind/vampire = pick(possible_vampires)
-		vampires += vampire
-		modePlayer += vampires
-		var/datum/mindslaves/slaved = new()
-		slaved.masters += vampire
-		vampire.som = slaved //we MIGT want to mindslave someone
-		vampire.restricted_roles = restricted_jobs
-		vampire.special_role = SPECIAL_ROLE_VAMPIRE
+		for(var/I in possible_vampires)
+			if(vampires.len >= secondary_enemies)
+				break
+			var/datum/mind/vampire = pick(possible_vampires)
+			vampires += vampire
+			modePlayer += vampires
+			possible_vampires -= vampire
+			var/datum/mindslaves/slaved = new()
+			slaved.masters += vampire
+			vampire.som = slaved //we MIGHT want to mindslave someone
+			vampire.restricted_roles = (restricted_jobs += secondary_restricted_jobs)
```

Comment:
This will keeping adding to the list needlessly and we'll wind up with something like `("Cyborg", "AI", "AI", "AI", ...)`
```suggestion
			vampire.restricted_roles = (restricted_jobs + secondary_restricted_jobs)
```